kvui.py

```diff
@@ -299,6 +300,39 @@ def apply_selection(self, rv, index, is_selected):
         self.selected = is_selected
 
 
+class AutocompleteHintInput(TextInput):
+    min_chars = NumericProperty(3)
+
```

Comment:
```suggestion
    text_validate_unfocus = False

```
Looks like this fixes the focusing issue. Note on its own, this would keep the focus on the hint search bar, which is desirable if the item name was invalid (so the player can correct a typo without retyping), or if the player wants to hint a lot of units from the same dropdown (say a ton of marine upgrades). If the hint was successful, it may be better to clear the text input, though.

If the desire is to reset the focus, I'd advise including this anyways but also moving the focus in `on_message()`. Something like `App.get_running_app().texinput.focus = True` felt fine when I tested it, and getting back to the hint textinput only took two shift+tabs. Without this change, the focus goes undefined and no amount of tabbing can get you back to the textinput.